src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -294,9 +294,8 @@ private BoundIndexerAccess TransformIndexerAccess(BoundIndexerAccess indexerAcce
 
             // Step three: Now fill in the optional arguments. (Dev11 uses the
             // getter for optional arguments in compound assignments.)
-            var getMethod = indexer.GetOwnOrInheritedGetMethod();
-            Debug.Assert((object)getMethod != null);
-            InsertMissingOptionalArguments(syntax, getMethod.Parameters, actualArguments);
+            var accessor = indexer.GetOwnOrInheritedGetMethod() ?? indexer.GetOwnOrInheritedSetMethod();
```

Comment:
I'd add a comment "but for deconstructions we use the setter if the getter is missing".